bellman/src/domain.rs

```diff
@@ -337,12 +341,12 @@ fn parallel_fft<E: ScalarEngine, T: Group<E>>(
                 let omega_step = omega.pow(&[(j as u64) << log_new_n]);
 
                 let mut elt = E::Fr::one();
-                for i in 0..(1 << log_new_n) {
+                for (i, tmp) in tmp.iter_mut().enumerate() {
```

Comment:
~~Previously this loop's body would execute  `1 << log_new_n` times. With this change it should execute `num_cpus` times according to the length of `tmp` when it's created above, right?. Won't that change the behavior?~~

Oh, I see `tmp`, is shadowed by the outermost loop so here it will have length `1 << log_new_n` and the behavior is unchanged. It might be worth renaming some variables to avoid similar confusion.